### ISSUE TITLE
Define stop-ship criteria

### ISSUE THREAD

**Issue by smlambert** [eclipse/openj9#11034]
Based on discussion in the OpenJ9 community call and retrospectively looking at the September release, it will be helpful to clearly define conditions or criteria that would constitute a 'stop-ship' of a release.

This issue is to capture a discussion and ultimately / hopefully result in helping both the OpenJ9 and AdoptOpenJDK projects during a release to decide on what actions to take based on the variety of scenarios that may occur during release period.

What constitutes a stop-ship event?  Here are some thoughts, to trigger the discussion...

- test failures that indicate a serious functional or performance problem
- test failures that require more investigation before knowing if they fall into the above type
- missing features that are intended to be part of a release but are not yet in (what amount of waiting period should this be)
- issues on a particular platform may stop the release for that platform while other platforms can be released (what amount of time to wait before deciding to make that call, if a fix for the platform issue can not be done in X days, then?)
- how to categorize/prioritize user raised issues (this is likely a 'gut feel' one, but throwing it in there anyway)

**Comment by andrewcraik** [comment]
I also think it is worth considering the stop-ship criteria for incubator features vs default features/modes. I agree we should always aim to keep the incubator quality as good as the rest of the implementation, but for stop-ship criteria should we have a slightly different criteria? Not sure of the answer but putting it out there.

**Comment by pshipton** [comment]
Looking at this from a different perspective, perhaps obvious, things that are non-stop ship are known issues that we've already decided to defer to another release. Unless something changes, like the frequency of the failure or what is impacted by the failure. In which case, it needs to be re-assessed.

**Comment by smlambert** [comment]
Right, an example of this would be in a [mac jdk11/jdk15 respin](https://github.com/AdoptOpenJDK/TSC/issues/188#issuecomment-736662219) today (to fix a mac installer issue), we see a single sanity.openjdk failure for openj9 which is intermittent and known (https://github.com/eclipse/openj9/issues/10580).  I decided this is a non-blocking failure (intermittent/known/raised in correct repo). 

What I had hoped from defining a stop-ship criteria, is to enable others to easily determine if something should or should not block a release.  Give others a set of rules to follow to make the determination.  

Someone would:
- see the failure in the release pipeline
- look at Deep History to see frequency
- look at Possible Issues list to see if its been reported
- if they find it has been reported (#10580), can they intuit that its not critical?  if issue does not have [blocker](https://github.com/eclipse/openj9/issues?q=is%3Aopen+is%3Aissue+label%3Ablocker) label is it safe to assume not blocking?  What about [prio:high](https://github.com/eclipse/openj9/issues?q=is%3Aissue+is%3Aopen+label%3Aprio%3Ahigh) label or [HOT](https://github.com/eclipse/openj9/issues?q=is%3Aopen+is%3Aissue+label%3AHOT) label (which does not seem used), do those labels hold particular meaning relating to blocking?

Ultimately, I guess someone may still need to post in the J9 Slack workspace to ask if its not clear whether to proceed, but again, who is to determine it?  Project leads, committers?

To @andrewcraik's point, I suppose this is for someone to know the release plan and if new features are present, know whether test failures are related to new features (in some cases this will be obvious, perhaps in other cases it will not be, but again, if we actively try to run green (disable failing tests as fixes are being worked), it will help the determination.

**Comment by pshipton** [comment]
- A simple rule we've used for many years is that something which fails 1/30 iterations of the test or more is a "blocking" (highest priority) problem and needs to be resolved asap. Typically this has been applied to JIT issues since VM issues have less tendency to be intermittent, but no reason it can't apply outside the JIT as well. In order to block a release, the failure also needs to have user impact, meaning the failure needs to occur using default options, or with options typically used by users. i.e. if it only fails 1/30 or more with `-Xjit:count=0` it's not a release blocking issue, as no user is expected to run with this option.

The rule gets a little fuzzy when the failure occurs only on some machines. Ideally when grinding a problem to determine failure rate it should be done across multiple machines to get a better understanding of real world failure rate. Not that we necessarily have a lot of different machines setups available, but we can at least test on what is available.

- Another factor is release frequency. If the next release is a month away, it's easier to justify shipping a known problem that will hopefully be fixed in the release one month later. If the next release is 3 months away, the impact of releasing a known problem needs to be more carefully considered.

**Comment by pshipton** [comment]
- about incubator features, a 1/30 or higher frequency problem could indeed be considered release blocking if the incubator feature is one of a small number of features in a new Java version. For example, if Java N+1 contains nothing more than an incubator feature, and that incubator feature doesn't work, there probably isn't much point in releasing Java N+1 until it does work.

**Comment by pshipton** [comment]
- as for who can determine if a problem is release blocking, I expect any committer is eligible to make a decision, or at least have an opinion, but it's up to each individual committer to decide if they are qualified to make that decision. It can be a tough decision, and a committer may want additional opinions from other committers so that a stop ship decision is more of a consensus. I don't expect the project leads necessarily have any special powers to decide for every type of problem, but take on the role of leading the decision making process if necessary.
